### DISCUSSION TITLE
Работа с selectRowList

### DISCUSSION THREAD

**Discussion by qye1** [totumonline/totum-ru-issues-and-discussions#401] [Как это сделать?] [answered]
Добрый день. Я с помощью функции selectRowList собираю из таблицы 2 значения: name и tf_qty. Мне нужно сложить все tf_qty, где одинаковый name и привести всё к виду:
Топор  — -15 штук
Нож  — -15 штук
Молоток  — 5 штук

Вот пример данных: [{"name":"Нож","tf_qty":"-10"},{"name":"Топор","tf_qty":"-10"},{"name":"Нож","tf_qty":"-5"},{"name":"Топор","tf_qty":"-10"},{"name":"Молоток","tf_qty":"10"},{"name":"Топор","tf_qty":"5"},{"name":"Молоток","tf_qty":"-5"}]

Но при этом количество элементов в данном списке может быть любым.

Подскажите пожалуйста, как это можно реализовать?

**Comment by erpic-ru** [comment, marked as answer]
Посмотрите тут: https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/154

**Reply by ttm-alexeypolunin** [reply]
Тимур прав здесь, я не досмотрел

**Comment by ttm-alexeypolunin** [comment]
Смотрите функцию rowCreateByLists: https://ru.docs.totum.online/functions#fn-rowCreateByLists

'''
example: rowCreateByLists(keys: $rowlist[[name]]; values: $rowlist[[tf_qty]])
    ~rowlist: selectRowList...
'''

двойные квадратные скобки - взять колонку по ключу
~ тилда обязательно чтоб запрос не повторять тк при повторении может получится разный порядок или размерность
